cdap-watchdog/src/test/java/co/cask/cdap/logging/write/LogCleanupTest.java

```diff
@@ -337,6 +351,184 @@ public void testRemoteLocationFactory() throws Exception {
+    addFilesModAfterRetention(fileMetaDataManager, deletionBoundary, dummyContext, nsContextDir, notToDelete);
+
+    LogCleanup logCleanup = new LogCleanup(fileMetaDataManager, rootLocationFactory, namespaceQueryAdmin,
+                                           namespacedLocationFactory, logBaseDir, RETENTION_DURATION_MS, impersonator);
+    final SetMultimap<String, Location> parentDirs = HashMultimap.create();
+    final Map<String, NamespaceId> namespaceIdMap = new HashMap<>();
+
+    for (int i = 0; i < 11; i++) {
+      logCleanup.cleanFilesWithoutMeta(deletionBoundary, namespaceIdMap, parentDirs, 15, 5);
+    }
+
+    // Assert only files without metadata is deleted.
+    for (Location location : toDelete) {
+      Assert.assertFalse("Location " + location + " is not deleted!", location.exists());
+    }
+
+    // Assert files with metadata is not deleted
+    for (Location location : notToDelete) {
+      Assert.assertFalse("Location " + location + " is deleted!", !location.exists());
+    }
+  }
+
+  @Test
+  public void testFilterLocationsWithMeta() throws Exception {
+    FileMetaDataManager fileMetaDataManager = injector.getInstance(FileMetaDataManager.class);
+    NamespacedLocationFactory namespacedLocationFactory = injector.getInstance(NamespacedLocationFactory.class);
+
+    // Deletion boundary
+    long deletionBoundary = System.currentTimeMillis() - RETENTION_DURATION_MS;
+    LOG.info("deletionBoundary = {}", deletionBoundary);
+
+    // Setup directories
+    LoggingContext dummyContext = new FlowletLoggingContext("ns", "app", "flw", "flwt", "run", "instance");
+    Location nsContextDir = createContextDir("ns", namespacedLocationFactory);
+    Set<Location> notToDelete = new HashSet<>();
+
+    for (int i = 0; i < 5; ++i) {
+      notToDelete.add(nsContextDir.append("2012-12-1" + i + "/del-1"));
+      notToDelete.add(nsContextDir.append("2012-12-1" + i + "/del-2"));
+      notToDelete.add(nsContextDir.append("2012-12-1" + i + "/del-3"));
+      notToDelete.add(nsContextDir.append("2012-12-1" + i + "/del-4"));
+    }
+
+    int counter = 0;
+    for (Location location : notToDelete) {
+      long modTime = deletionBoundary - counter - 10000;
+      fileMetaDataManager.writeMetaData(dummyContext, modTime,
+                                        createFile(location, modTime));
+      counter++;
+    }
+
+    LogCleanup logCleanup = new LogCleanup(fileMetaDataManager, rootLocationFactory, namespaceQueryAdmin,
+                                           namespacedLocationFactory, logBaseDir, RETENTION_DURATION_MS, impersonator);
+
+    logCleanup.filterLocationsWithMeta(new NamespaceId("ns"), notToDelete , 5);
+
+    // Assert files with metadata is not deleted
+    for (Location location : notToDelete) {
+      Assert.assertFalse("Location " + location + " is deleted!", !location.exists());
+    }
+    // delete these meta files
+    logCleanup.run();
```

Comment:
@poornachandra we need that because FileMetaDataManager is in Singleton scope which causes ``LogCleanupTest` to fail